java-checks/src/main/java/org/sonar/java/checks/regex/AbstractRegexCheck.java

```diff
@@ -94,12 +94,20 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
     }
     FlagSet flags = getFlags(mit);
     if (!flags.contains(Pattern.LITERAL)) {
-      getLiterals(args.get(0))
```

Comment:
As discussed, I feel that we should not change this logic and introduce some new overridable methods. It blurs the entry point of regex check, by providing many different ways to analyze regex.

I feel it would be better to keep AbstractRegexCheck having only `checkRegex()` being available for override, and unchanged in the way we use it. The rest of the methods being private or final. It makes all the rules consistent.